File: BinarytoDecimal.lua
#!/usr/bin/env lua
function BinarytoDecimal(binaryNumber)
	value = tostring(binaryNumber)
	length = string.len(value)
	output = 0
	for position = 0, length do
		if string.sub(value, position, position) == "1" then
			output = output + 2^(length - position)
		elseif string.sub(value, position, position) == "0" then

		end
	end
	return math.floor(output)
end

io.write("Binary Number: ")
num = io.read()
print(BinarytoDecimal(num))

File: BinarytoDecimal.dart
import 'dart:math';
import 'dart:io';

num BinarytoDecimal(num binaryNumber) {
    String value = binaryNumber.toString();
    num length = value.length;
    num output = 0;
    for (int position = 0; position < length; position++) {
        String ch = value.substring(position, position + 1);
        if (ch == '1') {
            output += pow(2, length - position - 1);
        } else if (ch == '0') {
            output += 0;
        } else {
            return -1;
        }
    }
    return output;
}

void main() {
    stdout.write("Binary Number: ");
    print(BinarytoDecimal(int.parse(stdin.readLineSync()!)));
}